### PR TITLE
fix: [CDS-45912]: nexus3 + ami trigger v2 added

### PR DESCRIPTION
**Pull request by satyam-harness**
### Summary
https://harness.atlassian.net/browse/CDS-45912
https://harness.atlassian.net/browse/CDS-46101

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by satyam-harness** [comment]
retrigger titlecheck

**Review by kapilkumawat86 — APPROVED** [review]

**Comment by satyam-harness** [comment]
retrigger jest

**Review by pankajsingh-harness — DISMISSED** [review]

**Review by pankajsingh-harness — DISMISSED** [review]

**Comment by satyam-harness** [comment]
retrigger cypress-cd

**Comment by satyam-harness** [comment]
retrigger cypress-cd

File: configs/restful-react.config.js [hunk 1]
```diff
@@ -26,6 +26,7 @@ module.exports = {
   },
   'cd-ng': {
     output: 'src/services/cd-ng/index.tsx',
+    file: 'src/services/cd-ng/swagger.json',
```
Comment (vikrant-harness) [inline configs/restful-react.config.js:29]: Do we need this?

File: configs/restful-react.config.js [hunk 2]
```diff
@@ -38,6 +39,7 @@ module.exports = {
   },
   'pipeline-ng': {
     output: 'src/services/pipeline-ng/index.tsx',
+    file: 'src/services/pipeline-ng/swagger.json',
```
Comment (vikrant-harness) [inline configs/restful-react.config.js:42]: here as well

**Review by vikrant-harness — APPROVED** [review]

**Review by vikrant-harness — APPROVED** [review]

File: src/modules/72-triggers/components/steps/ArtifactTriggerConfigPanel/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AmazonMachineImage/AmazonMachineImage.tsx [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+function FormComponent({
+  expressions,
+  prevStepData,
+  initialValues,
+  previousStep,
+  formik,
+  formClassName = ''
+}: any): React.ReactElement {
+  const { getString } = useStrings()
+  const [regions, setRegions] = React.useState<SelectOption[]>([])
```
Comment (PratyushGar) [inline src/modules/72-triggers/components/steps/ArtifactTriggerConfigPanel/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AmazonMachineImage/AmazonMachineImage.tsx:47]: regions state can be changed to a variable with useMemo on `regionData?.resource`

File: src/modules/72-triggers/components/steps/ArtifactTriggerConfigPanel/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AmazonMachineImage/AmazonMachineImage.tsx [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+  const { accountId, projectIdentifier, orgIdentifier } = useParams<ProjectPathProps>()
+  const [tags, setTags] = useState<SelectOption[]>([])
+
```
Comment (PratyushGar) [inline src/modules/72-triggers/components/steps/ArtifactTriggerConfigPanel/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AmazonMachineImage/AmazonMachineImage.tsx:50]: Same for tags

**Review by pankajsingh-harness — APPROVED** [review]

File: src/modules/72-triggers/components/steps/ArtifactTriggerConfigPanel/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AmazonMachineImage/AmazonMachineImage.tsx (hunk 3 above)
Comment (satyam-harness) [reply src/modules/72-triggers/components/steps/ArtifactTriggerConfigPanel/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AmazonMachineImage/AmazonMachineImage.tsx:47]: okay

**Pull request merged**